applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_solver.py

```diff
@@ -1,174 +1,96 @@
-                    "max_iteration": 400,
-                    "gmres_krylov_space_dimension": 100,
-                    "smoother_type":"ilu0",
-                    "coarsening_type":"ruge_stuben",
-                    "coarse_enough" : 5000,
-                    "krylov_type": "lgmres",
-                    "tolerance": 1e-9,
-                    "verbosity": 3,
-                    "scaling": false
-            }
-        }""")
-
-        # Overwrite the default settings with user-provided parameters.
-        self.settings = custom_settings
-        self.settings.ValidateAndAssignDefaults(default_settings)
-        model_part_name = self.settings["model_part_name"].GetString()
-
-        if model_part_name == "":
-            raise Exception('Please specify a model_part name!')
-
-        # This will be changed once the Model is fully supported!
-        if self.model.HasModelPart(model_part_name):
-            self.main_model_part = self.model[model_part_name]
-        else:
-            self.main_model_part = self.model.CreateModelPart(model_part_name)
-            domain_size = self.settings["domain_size"].GetInt()
-            if domain_size < 0:
-                raise Exception('Please specify a "domain_size" >= 0!')
-            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE, domain_size)
+                "solver_type": "amgcl"
+            },
+            "volume_model_part_name": "volume_model_part",
+            "skin_parts":[""],
+            "no_skin_parts": [""],
+            "move_mesh_flag": false
+        }''')
+
+        settings.ValidateAndAssignDefaults(default_settings)
+        return settings
+
+    def __init__(self, model, custom_settings):
+
+        super(PotentialFlowSolver, self).__init__(model, custom_settings)
+
+        # There is only a single rank in OpenMP, we always print
+        self._is_printing_rank = True
+
+        # Set the element and condition names for the replace settings
+        # TODO: Create a formulation class helper as soon as there is more than one element is present
+        self.element_name = "IncompressiblePotentialFlowElement"
+        self.condition_name = "PotentialWallCondition"
+        self.min_buffer_size = 1
 
         # Construct the linear solvers
         import KratosMultiphysics.python_linear_solver_factory as linear_solver_factory
         self.linear_solver = linear_solver_factory.ConstructSolver(self.settings["linear_solver_settings"])
 
-        print("Construction of LaplacianSolver finished")
-
     def AddVariables(self):
         # Degrees of freedom
         self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.VELOCITY_POTENTIAL)
         self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.AUXILIARY_VELOCITY_POTENTIAL)
 
         # Kratos variables
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
         self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
```

Comment:
Just for my understanding, does the order in which the variables are defined matter?

Reply:
no :)

Reply:
ok :+1: